File: Backend/AstroGame.Shared/Models/Players/PlayerSpeciesPerk.cs
using System;

namespace AstroGame.Shared.Models.Players
{
    public class PlayerSpeciesPerk
    {
        public Guid Id { get; set; }
        public Guid PlayerSpeciesId { get; set; }
        public Guid PerkId { get; set; }

        public PlayerSpecies PlayerSpecies { get; set; }
        public Perk Perk { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/players/player_species_perk.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/players/perk.dart';

import 'package:json_annotation/json_annotation.dart';

part 'player_species_perk.g.dart';

@GuidConverter()
@JsonSerializable()
class PlayerSpeciesPerk {
  late Guid id;
  late Guid playerSpeciesId;
  late Guid perkId;

  late Perk perk;

  PlayerSpeciesPerk();

  factory PlayerSpeciesPerk.fromJson(Map<String, dynamic> json) =>
      _$PlayerSpeciesPerkFromJson(json);
  Map<String, dynamic> toJson() => _$PlayerSpeciesPerkToJson(this);
}
